### PR TITLE
revert go 1.23 changes to fix pull-release-image-k8s-ci-builder

### PR DESCRIPTION
**Pull request by haitch**
#### What type of PR is this?

/kind regression

#### What this PR does / why we need it:

this pipeline breaks [pull-release-image-k8s-ci-builder](https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-release-image-k8s-ci-builder) , there are still go1.22 stack required, but when go.mod is based on go1.23, things break.

2 PRs caused the break:
- https://github.com/kubernetes/release/pull/3759 Bump the all group with 2 updates
- https://github.com/kubernetes/release/pull/3757 Switch to go 1.23

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by cpanato** [comment]
@haitch can you add the PR that those changes got in?

**Comment by haitch** [comment]
https://github.com/kubernetes/release/pull/3759 Bump the all group with 2 updates
https://github.com/kubernetes/release/pull/3757 Switch to go 1.23

**Review by cpanato — COMMENTED** [review]
/assign @saschagrunert

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module k8s.io/release
 
-go 1.23
+go 1.22.4
```
Comment (saschagrunert) [inline go.mod:3]: I assume that's the only change we need. Let's keep go 1.23 in GitHub actions to not revert the whole patch.

**Review by saschagrunert — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (cpanato) [reply go.mod:3]: agree

Comment (cpanato) [reply go.mod:3]: also maybe we can but to the latest go 1.22 as well

Comment (haitch) [reply go.mod]: no, it doesn't work: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/3781/pull-release-image-k8s-ci-builder/1844259485523120128

**Review by cpanato — APPROVED** [review]
i will follow up after this to update a few parts

/approve
/lgtm

/hold to defer final approval to @saschagrunert

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**